source/core/src/main/com/csse3200/game/areas/terrain/ShipPartTileComponent.java

```diff
@@ -0,0 +1,74 @@
+package com.csse3200.game.areas.terrain;
+
+import com.csse3200.game.components.Component;
+import com.csse3200.game.entities.Entity;
+import com.csse3200.game.entities.factories.ItemFactory;
+import com.csse3200.game.services.ServiceLocator;
+
+/**
+ * A tile that drops a ship part on destroy. Can contain a ship debris
+ * which must be destroyed first if it exists.
+ */
+public class ShipPartTileComponent extends Component {
+    private Entity shipDebris;
+
+    public ShipPartTileComponent() {
+        shipDebris = null;
+    }
+
+    @Override
+    public void create() {
+        entity.getEvents().addListener("destroy", this::destroyTile);
+    }
+
+    /**
+     * Set the ship debris associated with this tile.
+     *
+     * @param shipDebris entity to be added
+     */
+    public void addShipDebris(Entity shipDebris) {
+        this.shipDebris = shipDebris;
+        this.shipDebris.setPosition(entity.getPosition());
+        ServiceLocator.getEntityService().register(this.shipDebris);
+    }
+
+    /**
+     * A ship debris is present in this tile.
+     *
+     * @return true if there is ship debris present
+     */
+    private boolean containsShipDebris() {
+        return shipDebris != null;
+    }
+
+    /**
+     * Trigger the ship debris' destroy event and remove it
+     * from the tile.
+     */
+    private void destroyShipDebris() {
+        shipDebris.getEvents().trigger("destroy", null);
+        shipDebris = null;
+    }
+
+    /**
+     * Destroys the tile. If a ship debris is present, destroy it first.
+     * Drops a ship part on final destroy.
+     *
+     * @param tile the terrain tile that will become unoccupied once the ship
+     *             tile is destroyed.
+     */
+    private void destroyTile(TerrainTile tile) {
+        if (containsShipDebris()) {
+            destroyShipDebris();
+        } else {
+            // drop a ship part
+            Entity item = ItemFactory.createEgg(); // the ship is made of eggs apparently
+            item.setPosition(entity.getPosition());
+            ServiceLocator.getEntityService().register(item);
```

Comment:
does this work? might be better to call spawnEntityAt() or spawnEntity, same output but better :p

Reply:
yeah it does, I followed how we spawn items in other places.

idk if spawnEntityAt is cleaner, coz then we gotta convert from the player entity position to a map tile coordinate.

Reply:
hm, I dislike how we spawn it for peaceable destruction so was gonna change that, and there is a function in the GameMap that can convert, but it maybe just easier that way, just checking cause I don't wanna walk to pc, does spawn entity do anything else?

Reply:
spawnEntity does the same as spawnEntityAt just without setting the position

Reply:
it just feels a little silly to be converting from entity world pos -> tile pos -> world pos again in spawnEntityAt

Reply:
don't look at the code to hoe then lmaooo, but yeah I mean it looks fine with me and as long as it works, it works.